apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -439,6 +441,19 @@ const FormController = props => {
     const handleSuccessfulSubmit = data => {
       sessionStorage.removeItem(sessionStorageKey);
       onSuccessfulSubmit(data);
+
+      // Log application status change upon submission for Teacher Applications
+      if (window.location.href.includes('teacher')) {
+        const rp_requires_admin_approval =
+          regionalPartner.applications_principal_approval ===
+          'all_teachers_required';
+        analyticsReporter.sendEvent(EVENTS.APP_STATUS_CHANGE_EVENT, {
+          'application id': updatedApplicationId,
+          'application status': rp_requires_admin_approval
+            ? 'awaiting_admin_approval'
+            : 'unreviewed'
+        });
+      }
```

Comment:
For figuring out whether the user's application should be maraked as "awaiting_admin_approval" or "unreviewed" upon submission based on the regional partner choice, adding the event here seems to require the fewest file changes but also feels a bit clunky with checking the url to see if it's a teacher app and placing this inside the FormController rather than TeacherApplication. FormController just had access to the applicationId and form data (which TeacherApplication did not), so it would have required more changes to import that data into TeacherApplication just for an event log. But I am fully open to pushback if this doesn't sit right!

Reply:
I'm good with this –– thanks for logging your thought process!